fleet-server.yml

```diff
@@ -1,62 +1,31 @@
+# This config is intended to be used with a stand-alone fleet-server instance for development.
 output:
   elasticsearch:
-    hosts: '${ELASTICSEARCH_HOSTS:localhost:9200}'
+    hosts: '${ELASTICSEARCH_HOSTS:https://localhost:9200}'
     service_token: '${ELASTICSEARCH_SERVICE_TOKEN}'
+    ssl.ca_trusted_fingerprint: '${ELASTICSEARCH_CA_TRUSTED_FINGERPRINT}'
 
 fleet:
   agent:
-    id: 1e4954ce-af37-4731-9f4a-407b08e69e42
-    logging:
-      level: '${LOG_LEVEL:DEBUG}'
+    id: '${FLEET_SERVER_AGENT_ID:dev-fleet-server}'
 
-# Input config provided by the Elastic Agent for the server
-#inputs:
-#  - type: fleet-server
+inputs:
+  - type: fleet-server
+    policy.id: '${FLEET_SERVER_POLICY_ID:fleet-server-policy}'
 #    server:
-#      host: localhost
-#      port: 8220
-#      timeouts:
-#        checkin_long_poll: 300s # long poll timeout
-#      instrumentation:
-#        enabled: false
-#        hosts: ["localhost:8200"]
-#      profiler:
-#        enabled: true # enable profiler
-#      limits:
-#        policy_throttle: 100ms
-#        max_connetions: 150
-#          checkin_limit:
-#            interval: 100ms
-#            burst: 25
-#            max: 100
-#          artifact_limit:
-#            interval: 10ms
-#            burst: 5
-#            max: 10
-#          ack_limit:
-#            interval: 10ms
-#            burst: 20
-#            max: 10
-#          enroll_limit:
-#            interval: 50ms
-#            burst: 10
-#            max: 8
-#        ssl:
-#          enabled: true
-#          certificate: /creds/cert.pem
-#          key: /creds/key.pem
-#    cache:
-#      num_counters: 500000  # 10x times expected count
-#      max_cost: 50 * 1024 * 1024  # 50MiB cache size
+#      ssl:
+#        enabled: true
+#        certificate: /creds/cert.pem
+#        key: /creds/key.pem
+#        key_passphrase: /creds/key.pem
```

Comment:
I wasn't able to get an external agent operating correctly if the fleet-server does not have a cert (in http mode).
The agent could enroll but would be stuck in the updating state.

I think that was because the fleet UI requires an https url.
I can use and http url and the `--insecure` flag to enroll, the policy it retrieves would differ.
I don't think we really support fleet-server running as an http server and may want to emit a warning (mark as deprecated?) if we try to do so

Reply:
It would be nice though to have an easier way to start Fleet Server + Agent without a need to manage certs for development.